### PR TITLE
Add readability-implicit-bool-conversion

### PR DESCRIPTION
**Pull request by masterleinad**
Similar to https://github.com/dealii/dealii/pull/12262 let's just make `clang-tidy` also check for `int->bool` conversions.

### CODE REVIEW

**Comment by bangerth** [comment]
Unsurprisingly, this triggers hundreds of warnings :-)

**Comment by masterleinad** [comment]
> Unsurprisingly, this triggers hundreds of warnings :-)

Yes, I rather wanted to quickly see. Where we are at. Obviously, also not relevant for the release.

**Comment by bangerth** [comment]
I took another look at this. I would be in favor of merging the code changes for purposes of readability, and we can consider the changes to the clang-tidy flags separately. Want to drop the latter part and update the former and get this merged?

**Comment by masterleinad** [comment]
> I took another look at this. I would be in favor of merging the code changes for purposes of readability, and we can consider the changes to the clang-tidy flags separately. Want to drop the latter part and update the former and get this merged?

Sure, I'll have a look.

**Comment by masterleinad** [comment]
See https://github.com/dealii/dealii/pull/13150.

File: include/deal.II/matrix_free/fe_evaluation.h [hunk 1]
```diff
@@ -2980,7 +2980,7 @@ inline FEEvaluationBase<dim,
   Assert(this->data == nullptr, ExcInternalError());
   this->data =
     new internal::MatrixFreeFunctions::ShapeInfo<VectorizedArrayType>(
-      Quadrature<dim - is_face>(quadrature),
+      Quadrature<(is_face ? dim - 1 : dim)>(quadrature),
```
Comment (bangerth) [inline include/deal.II/matrix_free/fe_evaluation.h:2983]: nice, this is so much more readable!

**Review by bangerth — APPROVED** [review]

**Pull request merged**